File: src/main/java/discretemath/bool/table/TruthTableRow.java
package discretemath.bool.table;

import discretemath.common.BitString;

public interface TruthTableRow {

	int index();

	int argumentsCount();

	int functionsCount();

	BitString getArgumentsBitString();

	boolean getArgument(int index);

	FunctionValue getFunctionValue(int index);
}

File: src/main/scala/discretemath/bool/table/TruthTableRow.scala
package discretemath.bool.table;

import discretemath.common.BitString;

trait TruthTableRow {

  def index: Int

  def arguments: IndexedSeq[Boolean]

  def functionsCount: Int

  def getArgumentsBitString: BitString

  def getArgument(index: Int): Boolean

  def getFunctionValue(index: Int): FunctionValue
}